src/UsgsAstroFrameSensorModel.cpp

```diff
@@ -821,71 +824,68 @@ std::string UsgsAstroFrameSensorModel::constructStateFromIsd(const std::string&
 
         state["m_focalLength"] = focal_length;
         state["m_focalLengthEpsilon"] = epsilon;
+
+        std::cerr << "Focal Length Parsed!" << std::endl;
       }
 
-      // get sensor_location
+      // get sensor_position
       {
-        json jayson = isd.at("sensor_location");
-        json x = jayson.at("x");
-        json y = jayson.at("y");
-        json z = jayson.at("z");
+        json jayson = isd.at("sensor_position");
+        json positions = jayson.at("positions")[0];
+        json velocities = jayson.at("velocities")[0];
         json unit = jayson.at("unit");
 
-        state["m_currentParameterValue"] = json();
-        state["m_currentParameterValue"][0] = x;
-        state["m_currentParameterValue"][1] = y;
-        state["m_currentParameterValue"][2] = z;
-
         unit = unit.get<std::string>();
-        state["m_currentParameterValue"][0] = metric_conversion(state["m_currentParameterValue"][0].get<double>(), unit);
-        state["m_currentParameterValue"][1] = metric_conversion(state["m_currentParameterValue"][1].get<double>(), unit);
-        state["m_currentParameterValue"][2] = metric_conversion(state["m_currentParameterValue"][2].get<double>(), unit);
-      }
+        state["m_currentParameterValue"] = json();
+        state["m_currentParameterValue"][0] = metric_conversion(positions[0].get<double>(), unit);
+        state["m_currentParameterValue"][1] = metric_conversion(positions[1].get<double>(), unit);
+        state["m_currentParameterValue"][2] = metric_conversion(positions[2].get<double>(), unit);
+        state["m_spacecraftVelocity"] = velocities;
 
-      // get  sensor_velocity
-      {
-        json jayson = isd.at("sensor_velocity");
-        json x = jayson.at("x");
-        json y = jayson.at("y");
-        json z = jayson.at("z");
-
-        state["m_spacecraftVelocity"] = json();
-        state["m_spacecraftVelocity"][0] = x;
-        state["m_spacecraftVelocity"][1] = y;
-        state["m_spacecraftVelocity"][2] = z;
+        std::cerr << "Sensor Location Parsed!" << std::endl;
       }
 
       // get sun_position
+      // sun position is not strictly necessary, but is required for getIlluminationDirection.
       {
         json jayson = isd.at("sun_position");
-        json x = jayson.at("x");
-        json y = jayson.at("y");
-        json z = jayson.at("z");
+        json positions = jayson.at("positions")[0];
+        json unit = jayson.at("unit");
```

Comment:
Do we take in the sun velocities even though they are unused?

Reply:
We should. Right now this code isn't setup well for optional things. I'll look into a clean way to do this.

Reply:
Because they are currently unused, I think we should continue to ignore them. If we add support for them later as an optional value, then we should start bringing them in.

Reply:
:+1: